dfs/docker/docker.go

```diff
@@ -232,3 +233,28 @@ func (d *DockerClient) GetContainerStats(containerID string, timeout time.Durati
 
 	return result, nil
 }
+
+// FindImageByHash searches all local images for an image with the given hash
+func (d *DockerClient) FindImageByHash(imageHash string) (*dockerclient.Image, error) {
+	opts := dockerclient.ListImagesOptions{
+		All:     false,
```

Comment:
I was afraid that setting this to true would be too expensive, since this would compute hashes for all layers.  I will change this to be a passed-in parameter, and we will call it twice from FindImages:  Once set to false, then if we don't find it we will call it again with it set to true, saving the most expensive call for the last-ditch effort.